src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -197,10 +197,17 @@ private static unsafe int SysWrite(SafeSocketHandle handle, ReadOnlySpan<byte> b
             return sent;
         }
 
+        // The Linux kernel doesn't like it if we pass a null reference for buffer pointers, even if the length is 0.
+        // Replace any null pointer (e.g. from Memory<byte>.Empty) with a valid pointer.
+        private static ReadOnlySpan<byte> AvoidNullReference(ReadOnlySpan<byte> buffer) =>
+            Unsafe.IsNullRef(ref MemoryMarshal.GetReference(buffer)) ? Array.Empty<byte>() : buffer;
```

Comment:
buffer.IsEmpty would work,  too. That wouldn't mean it was a null ref,  but it wouldn't hurt to sub in Array.Empty for an empty span even if it's not null.